.github/workflows/type-check.yml

```
@@ -0,0 +1,39 @@
+    - uses: oven-sh/setup-bun@v2
+      with:
+        bun-version: 1.0.11
+    - run: bun install
+
+    - name: Check out the code
+      run: |
+        bun run type-check
+        bun run lint
+    
+    - name: Report type check errors
+      if: failure() && steps.type-check.outcome == 'failure'
```

actionlint (1.7.4)

property "type-check" is not defined in object type {}

(expression)